### PR TITLE
Added Brian Kernighan's algorithm.

### PR DESCRIPTION
**Pull request by debanjan-2002**
Sir,
I have added Brian Kernighan's algorithm in the Bit Manipulation section. I have explained it very briefly and given the reasoning and intuition behind the algorithm. Please review my pull request, and if you like how I explain then please consider merging it.
I have also been selected in GSSOC'21, and I am really into DSA. So it would be a really kind of you if you give me the opportunity to contribute in this amazing repository.

Regards,
Debanjan Poddar

### CODE REVIEW

**Comment by debanjan-2002** [comment]
#44

File: Code/C++/Brian_Kernighan.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+// Brian Kernighan's Algorithm
```
Comment (plazzy99) [inline Code/C++/Brian_Kernighan.cpp:4]: bring this comment on top of the file

File: Code/C++/Brian_Kernighan.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+/* TEST CASES
+
+    1) input   : 32
+       output  : 1
+    2) input   : 39
+       output  : 4
+
+    END...
```
Comment (plazzy99) [inline Code/C++/Brian_Kernighan.cpp:65]: Add time and space complexity

**Review by plazzy99 — CHANGES_REQUESTED** [review]
Only 1 issue to be solved in one or

Remove Sieve code and add another for that

**Comment by plazzy99** [comment]
Also, update the readme file for that

File: Code/C++/Brian_Kernighan.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// Brian Kernighan's Algorithm
+#include<bits/stdc++.h>
+using namespace std;
+
+
+/* It is an algorithm developed to count the number of set bits in a number
```
Comment (plazzy99) [inline Code/C++/Brian_Kernighan.cpp:6]: bring this comment outside the main code

**Review by plazzy99 — CHANGES_REQUESTED** [review]

**Comment by plazzy99** [comment]
Also please link your pr with the issue

**Pull request closed without merging**